packages/language/core/v1/src/contract.ts

```diff
@@ -219,8 +256,9 @@ export type Timeout = bigint;
 export const TimeoutGuard: t.Type<Timeout> = t.bigint;
 
 /**
- * @hidden
+ * @experimental
  */
+// DISCUSSION: I think this should be renamed dateToPOSIX and moved to the time module in the adapter package
```

Comment:
yes it should ! It is in Domain Driven Design what we call mappers from the "anti-corruption layer", mappers between core domains and the other domains types

Reply:
Could we think about something more structured ? a Mapper Class or something?  `Mapper<Marlowe.Core.Timeout,PosixTime>` with a `from` and a `to`

Reply:
Doing functional programming and having fp-ts and io-ts, we could use `iso` as a structured way. Moreover we could have a simple `POSIXTime` class that has a `toJSON` method the serializes as bigint, and make the `io-ts` decoder to decode from bigint, but I think the two isolated functions are easier to understand